icu4c/source/common/brkeng.h

```diff
@@ -286,13 +288,15 @@ class ICULanguageBreakFactory : public LanguageBreakFactory {
   */
   virtual const LanguageBreakEngine *getEngineFor(UChar32 c, const char* locale) override;
 
+#if !UCONFIG_NO_SERVICE
```

Comment:
In public header files / class definitions, I don't think we can conditionally hide a virtual method -- that would make the vtable incompatible.
But I guess it's ok in an internal header file, since with such a switch one would build a different library anyway?

Reply:
i guess https://github.com/unicode-org/icu/blob/maint/maint-74/icu4c/source/common/unicode/rbbi.h#L805-L815 is a static.